stix/utils/nsparser.py

```diff
@@ -337,208 +337,120 @@ def collect(self, entity):
+NS_STIX_EXT_TEST_OPENIOC = Namespace('http://stix.mitre.org/extensions/TestMechanism#OpenIOC2010-1', 'stix-openioc', 'http://stix.mitre.org/XMLSchema/extensions/test_mechanism/open_ioc_2010/1.2/open_ioc_2010_test_mechanism.xsd')
+NS_STIX_EXT_TEST_SNORT = Namespace('http://stix.mitre.org/extensions/TestMechanism#Snort-1', 'snortTM', 'http://stix.mitre.org/XMLSchema/extensions/test_mechanism/snort/1.2/snort_test_mechanism.xsd')
+NS_STIX_EXT_TEST_YARA = Namespace('http://stix.mitre.org/extensions/TestMechanism#YARA-1', 'yaraTM', 'http://stix.mitre.org/XMLSchema/extensions/test_mechanism/yara/1.2/yara_test_mechanism.xsd')
+NS_STIX_EXT_VULN_CVRF = Namespace('http://stix.mitre.org/extensions/Vulnerability#CVRF-1', 'stix-cvrf', 'http://stix.mitre.org/XMLSchema/extensions/vulnerability/cvrf_1.1/1.2/cvrf_1.1_vulnerability.xsd')
+
+#: Namespaces used by external schemas that are incorporated in STIX.
+NS_CAPEC = Namespace('http://capec.mitre.org/capec-2', 'capec', '')
+NS_MAEC_PACKAGE = Namespace('http://maec.mitre.org/XMLSchema/maec-package-2', 'maecPackage', '')
+NS_OVAL_DEF = Namespace('http://oval.mitre.org/XMLSchema/oval-definitions-5', 'oval-def', '')
+NS_OVAL_VAR = Namespace('http://oval.mitre.org/XMLSchema/oval-variables-5', 'oval-var', '')
+NS_OPENIOC = Namespace('http://schemas.mandiant.com/2010/ioc', 'ioc', '')
+NS_OPENIOC_TR = Namespace('http://schemas.mandiant.com/2010/ioc/TR/', 'ioc-tr', '')
+NS_CVRF = Namespace('http://www.icasi.org/CVRF/schema/cvrf/1.1', 'cvrf', '')
+# NOTE: These three schemas are hosted on the STIX site despite them being
+# defined outside of STIX
+NS_OASIS_CIQ_XAL = Namespace('urn:oasis:names:tc:ciq:xal:3', 'xal', 'http://stix.mitre.org/XMLSchema/external/oasis_ciq_3.0/xAL.xsd')
+NS_OASIS_CIQ_XPIL = Namespace('urn:oasis:names:tc:ciq:xpil:3', 'xpil', 'http://stix.mitre.org/XMLSchema/external/oasis_ciq_3.0/xPIL.xsd')
+NS_OASIS_CIQ_XNL = Namespace('urn:oasis:names:tc:ciq:xnl:3', 'xnl', 'http://stix.mitre.org/XMLSchema/external/oasis_ciq_3.0/xNL.xsd')
+
+
+STIX_NAMESPACES = NamespaceSet()
+EXTENSION_NAMESPACES = NamespaceSet()
+
+# Magic to automatically register all Namespaces defined in this module.
+for k, v in dict(globals()).items():
+    if k.startswith('NS_'):
+        register_namespace(v)
+        if k.startswith('NS_MARKING_') or k.startswith('NS_STIX_'):
+            STIX_NAMESPACES.add(v)
+        else:
+            EXTENSION_NAMESPACES.add(v)
 
-    DEFAULT_STIX_NAMESPACES.update(ns_to_prefix)
-    DEFAULT_STIX_PREFIX_TO_NAMESPACE.update(prefix_to_ns)
-    DEFAULT_STIX_NAMESPACES_TUPLE = tuple(DEFAULT_STIX_NAMESPACES.iterkeys())
-    DEFAULT_STIX_SCHEMALOCATIONS.update(ns_to_schemalocation)
+
+#: Default namespace->alias mappings. These can be overriden by user-provided dictionaries on export.
+DEFAULT_STIX_NS_TO_PREFIX = dict((x.name, x.prefix) for x in STIX_NAMESPACES)
+# NOTE: CybOX Core and Common are no longer in this mapping. Is this OK?
```

Comment:
`STIX_NAMESPACES`  is just the stix.mitre.org and data-marking.mitre.org namespaces (not CybOX, MAEC, Extensions, etc.). It's used in the STIX core binding, otherwise we can just get rid of it entirely (I didn't look into how it's used there.)